hal/mbed_ticker_api.c

```diff
@@ -74,15 +74,17 @@ void ticker_insert_event(const ticker_data_t *const data, ticker_event_t *obj, t
         prev = p;
         p = p->next;
     }
+    
+    /* if we're at the end p will be NULL, which is correct */
+    obj->next = p;
```

Comment:
> it's still not a good idea to leave a global resource in a corrupted state before calling an external function.

100% agreed. 

> Observe that this is in the generic HAL and us_ticker_set_interrupt is KL25Z specific. So "if us_ticker_set_interrupt does nothing more than set the interrupt" is only true for KL25Z and only after this change

It is partially true: `us_ticker_set_interrupt` is indeed not supposed to call `us_ticker_irq_handler` but  it is difficult to write compliant function because the description of hal functions doesn't cover/define details such as side effect allowed by implementation or expected behaviours for corner cases.